CMakeLists.txt

```diff
@@ -100,7 +100,9 @@ endif()
 
 # --- Linux Joystick API -------------------------------------------------------
 
-check_include_files(linux/joystick.h HAVE_LINUX_JOYSTICK_H)
+if(NOT CORE_SYSTEM_NAME STREQUAL android)
```

Comment:
because it's highly unlikely that any other platform then linux provides this.
Checking for linux here would also bring it in line with the other platform checks above.